charts/temporal/templates/_helpers.tpl

```diff
@@ -71,6 +71,41 @@ Define the AppVersion
 {{- end -}}
 {{- end -}}
 
+{{/*
+Create the annotations for all resources
+*/}}
+{{- define "temporal.resourceAnnotations" -}}
+{{- $global := index . 0 -}}
+{{- $scope := index . 1 -}}
+{{- $resourceType := index . 2 -}}
+{{- $component := "server" -}}
+{{- if (or (eq $scope "admintools") (eq $scope "web")) -}}
+{{- $component = $scope -}}
+{{- end -}}
+{{- with $resourceType -}}
+{{- $resourceTypeKey := printf "%sAnnotations" . -}}
+{{- $componentAnnotations := (index $global.Values $component $resourceTypeKey) -}}
+{{- $scopeAnnotations := dict -}}
+{{- if hasKey (index $global.Values $component) $scope -}}
+{{- $scopeAnnotations = (index $global.Values $component $scope $resourceTypeKey) -}}
+{{- end -}}
+{{- $resourceAnnotations := merge $scopeAnnotations $componentAnnotations -}}
+{{- range $annotation_name, $annotation_value := $resourceAnnotations }}
+{{ $annotation_name }}: {{ $annotation_value }}
+{{- end -}}
+{{- end -}}
+{{ include "temporal.additionalResourceAnnotations" $global }}
+{{- end -}}
+
+{{/*
+Additonal user specified annotations for all resources
```

Comment:
Couldn't this just be inlined above? It's never called on it's own.

Reply:
Just noticed the label version is split out. That should be inlined also, but don't worry about that for this PR.

Reply:
It was indeed intended to align with the approach on the labels. But agree that inline is better, because it's not called on it's own.